src/components/pages/address-overview.tsx

```diff
@@ -115,41 +121,40 @@ const AddressOverview: React.FC<{ address?: Address }> = ({ address }) => {
               setIsModalOpen={setIsMarkAsPrimaryModalOpen}
               onConfirmationPress={async () => {
                 await fetchJson(
-                  `/api/profile/addresses/${address?.id}/mark-as-primary/`,
+                  `/api/profile/addresses/${addressToDisplay?.id}/mark-as-primary/`,
                   "POST",
                   {},
-                )
-                  .then(() => {
-                    const modifiedAddresses = [
-                      ...(addresses || ([] as Address[])),
-                    ];
-                    const oldMarkedAddress = modifiedAddresses.find(
-                      (address) => address.primary,
-                    );
-                    modifiedAddresses.forEach(
-                      (address) => (address.primary = false),
+                ).then(() => {
+                  const modifiedAddresses = [
+                    ...(addresses || ([] as Address[])),
+                  ];
+                  const oldMarkedAddress = modifiedAddresses.find(
+                    (address) => address.primary,
+                  );
+
+                  modifiedAddresses.forEach(
+                    (address) => (address.primary = false),
+                  );
+
+                  const addressToMark = modifiedAddresses.find(
+                    ({ id }) => id === addressToDisplay.id,
+                  );
+
+                  if (addressToMark && oldMarkedAddress) {
+                    addressToMark.primary = true;
+                    mutate("/api/profile/addresses/", modifiedAddresses);
+                    mutate(
+                      `/api/profile/addresses/${oldMarkedAddress.id}/`,
+                      oldMarkedAddress,
                     );
-                    const addressToMark = modifiedAddresses.find(
-                      (a) => a.id === address.id,
+                    mutate(
+                      `/api/profile/addresses/${addressToMark.id}/`,
+                      oldMarkedAddress,
```

Comment:
Shouldn't it be `addressToMark` here ? 🤔

Reply:
I did not touch on this part, as the PR was already pretty big, but I'll take a look at it.

Reply:
I may have missed something, but I don't understand this part as it is 🤨  why mutate the new address marked as primary with the old one value ?

Reply:
I have also a hard time getting it. I asked if a User could have multiple primary address on Connect.Profile, and if it's not the case, I will simplify the implementation.

Reply:
Done. I will refactor some more in an other PR to remove the unnecessary fetch of the current address.